src/operator/share.ts

```diff
@@ -2,6 +2,16 @@ import {Observable} from '../Observable';
 import {multicast} from './multicast';
 import {Subject} from '../Subject';
 
+/**
+ * Returns a new Observable that multicasts (shares) the original Observable. As long as there is at least one
+ * Subscriber this Observable will be subscribed and emitting data. When all subscribers have unsubscribed it will
+ * unsubscribe from the source Observable. Because the Observable is multicasting it makes the stream `hot`.
+ * This is an alias for .publish().refCount().
+ *
+ * <img src="./img/share.png" width="100%">
+ *
+ * @returns {Observable<T>} an Observable that upon connection causes the source Observable to emit items to its Observers
+ */
```

Comment:
This comment should be places above the `share` function, not above the `shareSubjectFactory`.